client/jetpack-connect/auth-logged-in-form.jsx

```diff
@@ -185,6 +197,17 @@ class LoggedInForm extends Component {
 		return startsWith( get( props, [ 'authorizationData', 'queryObject', 'from' ] ), 'jpo' );
 	}
 
+	shouldRedirectJetpackStart( { partnerId } ) {
+		const partnerRedirectFlag = config.isEnabled(
+			'jetpack/connect-redirect-pressable-credential-approval'
+		);
+
+		// If the redirect flag is set, then we conditionally redirect the Pressable client to
+		// a credential approval screen. Otherwise, we need to redirect all other partners back
+		// to wp-admin.
+		return partnerRedirectFlag ? partnerId && PRESSABLE_PARTNER_ID !== partnerId : partnerId;
```

Comment:
It's taken me several readings to appreciate what's going on here. This should return `true` if we have a `partnerId` and it's _not_ the `PRESSABLE_PARTNER_ID`, otherwise `false`. Is that right?

Do you think something like the following better expresses the intent?

```js
shouldRedirectJetpackStart( { partnerId } ) {
  if ( ! partnerId ) {
    return false;
  }

  // Pressable sites have a special NUX behind a flag
  if ( config.isEnabled( 'jetpack/connect-redirect-pressable-credential-approval' ) ) {
    return PRESSABLE_PARTNER_ID !== partnerId
  }

  return true;
}
```

Reply:
Thanks for that. I'd prefer to merge for now, but will be glad to address in a follow-up PR.